### PR TITLE
sidecars: Reuse sidecar-shim skeleton code

### PR DESCRIPTION
**Pull request by ormergi**
**What this PR does / why we need it**:
The sidecar-shim main function has additional logic that solve issues with migration that other sidecar implementations may suffer from (e.g.: slirp & passt sidecars).

This PR export the sidecar-shim main function to new package 'launcher'.
The new package exposes the Run() function:
It exposes an entrypoint for registering callback server
which is the difference between sidecar implementations.

With the new launcher package we provide single skeleton for all sidecar implementations (w or w/o sidecar shim layer) and streamline sidecar developing experience a bit.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Checklist**

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by ormergi** [comment]
> /cc

I was just about to look up your github nick and assign you :smile:

File: cmd/sidecars/launcher/launcher.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
+package launcher
```
Comment (victortoso) [inline cmd/sidecars/launcher/launcher.go:1]: I'm wondering about the name. If we want to share more code in the future between sidecar-shim and other sidecars, launcher could be odd but it is perfect for now... Naming is hard.

**Review by victortoso — COMMENTED** [review]
Looks fine to me. I'd suggest to take the opportunity and implement v1alpha3 for both network pluggins too, as that solves a lifecycle [issue](https://github.com/kubevirt/kubevirt/issues/8395).

File: cmd/sidecars/launcher/launcher.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
```
Comment (victortoso) [inline cmd/sidecars/launcher/launcher.go:2]: Should probably add the copyright note too.

**Pull request closed without merging**